platform/ui/src/components/downloadDialog/DownloadDialog.js

```diff
@@ -0,0 +1,235 @@
+import './DownloadDialog.styl';
+
+import React, { PureComponent } from 'react';
+import { withTranslation } from '../../utils/LanguageProvider';
+import Modal from 'react-bootstrap-modal';
+import 'react-bootstrap-modal/lib/css/rbm-patch.css';
+import PropTypes from 'prop-types';
+
+class DownloadDialog extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: this.props.isOpen,
+      width: 3000,
+      height: 3000,
+      fileName: '',
+      fileType: 'png',
+      showAnnotations: true,
+      fileTypeOptions: [
+        {
+          key: 'jpg',
+          value: 'jpg'
+        },
+        {
+          key: 'png',
+          value: 'png'
+        }
+      ],
+      previewElementRef: null
+    };
+
+    this.submitForm = this.submitForm.bind(this);
+    this.onClose = this.onClose.bind(this);
+  }
+
+  submitForm(e) {
+    e.preventDefault();
+
+    const formData = {
+      width: this.state.width,
+      height: this.state.height,
+      fileName: this.state.fileName,
+      fileType: this.state.fileType,
+      showAnnotations: this.state.showAnnotations,
+    };
+
+    const thereIsAnyEmpty = Object.values(formData).some(field => field === '');
+
+    if (thereIsAnyEmpty) {
+      alert('Please full fill all fields.');
+      return false;
+    }
+
+    this.props.takeAndDownloadSnapShot(formData);
+
+    return false;
+  };
+
+  static propTypes = {
+    isOpen: PropTypes.bool.isRequired,
+    toggleDownloadDialog: PropTypes.func.isRequired,
+    takeAndDownloadSnapShot: PropTypes.func.isRequired,
+    mountPreview: PropTypes.func.isRequired,
+    onResize: PropTypes.func.isRequired,
+    toggleAnnotations: PropTypes.func.isRequired,
+    forceRenderUpdate: PropTypes.func.isRequired,
+    setCacheReferences: PropTypes.func.isRequired,
+    cleanViewPortClone: PropTypes.func.isRequired,
```

Comment:
This is going into a "generic" component library. Some of these props feel like they're very specific to our implementation.

Reply:
Please review. Props updated.